packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -724,30 +720,168 @@ class TimeInput extends React.Component {
       { 'initial-focus': this.state.secondInitialFocused },
     ]);
 
-    const formatDescriptionId = `terra-time-input-description-format-${this.uuid}`;
-
-    const format = showSeconds
-      ? `(${intl.formatMessage({ id: 'Terra.timeInput.hh' })}:${intl.formatMessage({ id: 'Terra.timeInput.mm' })}:${intl.formatMessage({ id: 'Terra.timeInput.ss' })})`
-      : `(${intl.formatMessage({ id: 'Terra.timeInput.hh' })}:${intl.formatMessage({ id: 'Terra.timeInput.mm' })})`;
+    function mask() {
+      if (showSeconds) {
+        return intl.formatMessage({
+          id: 'Terra.timeInput.maskHourMinute',
+          defaultMessage: '(hh:mm:ss)',
+          description: 'A visual hint that the Time Input expects two-digit hour, minute and second.',
+        });
+      }
+      return intl.formatMessage({
+        id: 'Terra.timeInput.maskHourMinuteSecond',
+        defaultMessage: '(hh:mm)',
+        description: 'Like Terra.timeInput.maskHourMinute but no second.',
+      });
+    }
+
+    /**
+     * Returns a localized description for the hour input that reflects whether this is a 12 or 24-hour clock.
+     *
+     * NOTE: the description does not take the place of validation or error messages.
+     */
+    function hourDescription() {
+      if (is12Hour) {
+        return intl.formatMessage({
+          id: 'Terra.timeInput.hourDescriptionTwelve',
+          defaultMessage: 'A two-digit 12-hour value',
+          description: `Explains to screen reader users that the hour field needs a two digit hour. This will be read
+          only when a screen reader is enabled. It is never displayed. It will be read when the user has focused on
+          the hour input or when the screen reader is reading the page to the user.`,
+        });
+      }
+      return intl.formatMessage({
+        id: 'Terra.timeInput.hourDescriptionTwentyFour',
+        defaultMessage: 'A two-digit 24-hour value',
+        description: 'Like Terra.timeInput.hourDescriptionTwelve but for a 24-hour clock.',
+      });
+    }
+
+    // Fan out some component-level props into input-level a11y attributes. See the big comment below for more info.
+    inputAttributes.isInvalid = isInvalid;
+    inputAttributes.disabled = disabled;
+    inputAttributes.required = required;
+
+    const a11yString = TimeUtil.getA11YTimeValue(this.props, this.state, this.postMeridiem);
+
+    const groupLabel = intl.formatMessage({ // fix
```

Comment:
Is there still something that needs fixed here?

Reply:
Should this just use the provided "label" as is and not need to do any translation?